Assets/Documentation/man/man3/GazeGuidingPathPlayerSecondPath.3
.TH "GazeGuidingPathPlayerSecondPath" 3 "Version 0.1" "NPP VR Simulation" \" -*- nroff -*-
.ad l
.nh
.SH NAME
GazeGuidingPathPlayerSecondPath \- This class implements logic for gaze guiding in a virtual environment\&.  

.SH SYNOPSIS
.br
.PP
.PP
Inherits MonoBehaviour\&.
.SS "Public Member Functions"

.in +1c
.ti -1c
.RI "void \fBArrow3D\fP ()"
.br
.RI "This method adds a continuous rotating 3D arrow that is rendered above an exact rotary switch to indicate the direction in which the player is supposed to turn the switch\&. "
.ti -1c
.RI "void \fBArrow3DBinaer\fP ()"
.br
.RI "This method adds a binary rotating 3D arrow rendered above a binary rotary switch to indicate the direction in which the player is supposed to turn the switch\&. "
.ti -1c
.RI "void \fBRemoveArrow3D\fP ()"
.br
.RI "This method removes a rotating 3D arrow or 3DBinary arrow that is currently being rendered above an exact or a binary rotary switch\&. "
.ti -1c
.RI "void \fBTriggerTarget\fP (\fBGazeGuidingTarget\fP target, bool Flip3DArrow=false)"
.br
.ti -1c
.RI "void \fBTriggerTargetNAME\fP (string targetName, \fBGazeGuidingTarget\&.TargetType\fP type, bool Flip3DArrow=false)"
.br
.RI "This method sets the current target of a \fBGazeGuidingPathPlayerSecondPath\fP instance\&. "
.ti -1c
.RI "void \fBClearLine\fP ()"
.br
.RI "This method clears the scene of all visual aids that are currently being rendered\&. "
.in -1c
.SS "Public Attributes"

.in +1c
.ti -1c
.RI "bool \fBDirectionArrowEnabled\fP = true"
.br
.RI "
.PP
\fBParameters\fP
.RS 4
\fIDirectionArrowEnabled\fP toggles a directional arrow indicating a direction the player should turn to
.RE
.PP
"
.ti -1c
.RI "bool \fBArrow3DEnabled\fP = true"
.br
.RI "
.PP
\fBParameters\fP
.RS 4
\fIArrow3DEnabled\fP toggles a 3D arrow indicating the direction an exact rotary switch should be turned in
.RE
.PP
"
.ti -1c
.RI "bool \fBArrow3DBinearEnabled\fP = true"
.br
.RI "
.PP
\fBParameters\fP
.RS 4
\fIArrow3DBinearEnabled\fP toggles 3D arrow indicating the direction a binary rotary switch should be turned in
.RE
.PP
"
.ti -1c
.RI "List< \fBGazeGuidingTarget\fP > \fBtargets\fP"
.br
.RI "
.PP
\fBParameters\fP
.RS 4
\fIDirectionArrowOnScreen\fP toggles the on screen display of the directional arrow
.RE
.PP
"
.ti -1c
.RI "float \fBpathDisplayDistance\fP = 5\&.0f"
.br
.RI "
.PP
\fBParameters\fP
.RS 4
\fIpathDisplayDistance\fP specifies the distance between the player and an animated path that is displayed as visual cue 
.RE
.PP
"
.ti -1c
.RI "float \fBanimationDuration\fP = 1\&.0f"
.br
.RI "
.PP
\fBParameters\fP
.RS 4
\fIanimationDuration\fP specifies the duration of the path animation
.RE
.PP
"
.ti -1c
.RI "Material \fBlineMaterial\fP"
.br
.RI "
.PP
\fBParameters\fP
.RS 4
\fIlineMaterial\fP is a Material used to render a path as visual cue
.RE
.PP
"
.in -1c
.SS "Private Member Functions"

.in +1c
.ti -1c
.RI "void \fBStart\fP ()"
.br
.RI "This method initialises the \fBGazeGuidingPathPlayerSecondPath\fP\&. "
.ti -1c
.RI "void \fBUpdate\fP ()"
.br
.RI "<summaQry> This method updates all active gaze-guiding tools\&. "
.ti -1c
.RI "IEnumerator \fBAnimatePath\fP (Vector3 targetPosition)"
.br
.RI "This method animates a path of arrows guiding the player towards the component set as currentTarget of the \fBGazeGuidingPathPlayerSecondPath\fP\&. "
.in -1c
.SS "Private Attributes"

.in +1c
.ti -1c
.RI "\fBGazeGuidingTarget\fP \fBcurrentTarget\fP"
.br
.RI "
.PP
\fBParameters\fP
.RS 4
\fIcurrentTarget\fP is the currently active \fBGazeGuidingTarget\fP for this \fBGazeGuidingPathPlayerSecondPath\fP
.RE
.PP
"
.ti -1c
.RI "LineRenderer \fBlineRenderer\fP"
.br
.RI "
.PP
\fBParameters\fP
.RS 4
\fIlineRenderer\fP is a LineRenderer object used to render a path as visual cue
.RE
.PP
"
.ti -1c
.RI "Coroutine \fBanimatePathCoroutine\fP"
.br
.RI "
.PP
\fBParameters\fP
.RS 4
\fIanimatePathCoroutine\fP is a Coroutine used to animate a path as visual cue
.RE
.PP
"
.ti -1c
.RI "bool \fBisAnimating\fP = false"
.br
.RI "
.PP
\fBParameters\fP
.RS 4
\fIisAnimating\fP tracks whether a path is currently being animated as visual cue
.RE
.PP
"
.ti -1c
.RI "GameObject \fBarrow3DPrefab\fP"
.br
.RI "
.PP
\fBParameters\fP
.RS 4
\fIarrow3DPrefab\fP is a reference to an Arrow3D prefab used to render continuous rotational 3DArrows
.RE
.PP
"
.ti -1c
.RI "GameObject \fBarrow3DInstance\fP"
.br
.RI "
.PP
\fBParameters\fP
.RS 4
\fIarrow3DInstance\fP is an instance of a Arrow3D prefab used to render continuous rotational 3DArrows<param> 
.RE
.PP
"
.ti -1c
.RI "GameObject \fBarrow3DBinaerPrefab\fP"
.br
.RI "
.PP
\fBParameters\fP
.RS 4
\fIarrow3DBinaerPrefab\fP is a reference to an Arrow3D prefab used to render binary rotational 3DArrows
.RE
.PP
"
.ti -1c
.RI "GameObject \fBarrow3DBinaerInstance\fP"
.br
.RI "
.PP
\fBParameters\fP
.RS 4
\fIarrow3DBinaerInstance\fP is an instance of an Arrow3D prefab used to render binary rotational 3DArrows<param> 
.RE
.PP
"
.ti -1c
.RI "bool \fBarrow3DInstanceCreated\fP = false"
.br
.RI "
.PP
\fBParameters\fP
.RS 4
\fIarrow3DInstanceCreated\fP is a flag tracking whether an Arrow3D instance has been created 
.RE
.PP
"
.ti -1c
.RI "bool \fBarrow3DBinaerInstanceCreated\fP = false"
.br
.RI "
.PP
\fBParameters\fP
.RS 4
\fIarrow3DBinaerInstanceCreated\fP is a flag tracking whether an Arrow3DBinary instance has been created 
.RE
.PP
"
.in -1c
.SH "Detailed Description"
.PP 
This class implements logic for gaze guiding in a virtual environment\&. 
.SH "Member Function Documentation"
.PP 
.SS "IEnumerator GazeGuidingPathPlayerSecondPath\&.AnimatePath (Vector3 targetPosition)\fR [private]\fP"

.PP
This method animates a path of arrows guiding the player towards the component set as currentTarget of the \fBGazeGuidingPathPlayerSecondPath\fP\&. 
.PP
\fBParameters\fP
.RS 4
\fItargetPosition\fP is a Vector3 position of the endpoint of the path 
.RE
.PP

.SS "void GazeGuidingPathPlayerSecondPath\&.Arrow3D ()"

.PP
This method adds a continuous rotating 3D arrow that is rendered above an exact rotary switch to indicate the direction in which the player is supposed to turn the switch\&. 
.SS "void GazeGuidingPathPlayerSecondPath\&.Arrow3DBinaer ()"

.PP
This method adds a binary rotating 3D arrow rendered above a binary rotary switch to indicate the direction in which the player is supposed to turn the switch\&. 
.SS "void GazeGuidingPathPlayerSecondPath\&.ClearLine ()"

.PP
This method clears the scene of all visual aids that are currently being rendered\&. 
.SS "void GazeGuidingPathPlayerSecondPath\&.RemoveArrow3D ()"

.PP
This method removes a rotating 3D arrow or 3DBinary arrow that is currently being rendered above an exact or a binary rotary switch\&. 
.SS "void GazeGuidingPathPlayerSecondPath\&.Start ()\fR [private]\fP"

.PP
This method initialises the \fBGazeGuidingPathPlayerSecondPath\fP\&. 
.SS "void GazeGuidingPathPlayerSecondPath\&.TriggerTarget (\fBGazeGuidingTarget\fP target, bool Flip3DArrow = \fRfalse\fP)"

.SS "void GazeGuidingPathPlayerSecondPath\&.TriggerTargetNAME (string targetName, \fBGazeGuidingTarget\&.TargetType\fP type, bool Flip3DArrow = \fRfalse\fP)"

.PP
This method sets the current target of a \fBGazeGuidingPathPlayerSecondPath\fP instance\&. 
.PP
\fBParameters\fP
.RS 4
\fItargetName\fP contains the name of a rotary switch that is the new target for the \fBGazeGuidingPathPlayerSecondPath\fP 
.br
\fItype\fP is an Enum specifying the type of rotary switch passed in targetName 
.br
\fIFlip3DArrow\fP toggles whether 3D arrows rotate clockwise 
.RE
.PP

.SS "void GazeGuidingPathPlayerSecondPath\&.Update ()\fR [private]\fP"

.PP
<summaQry> This method updates all active gaze-guiding tools\&. 
.SH "Member Data Documentation"
.PP 
.SS "Coroutine GazeGuidingPathPlayerSecondPath\&.animatePathCoroutine\fR [private]\fP"

.PP

.PP
\fBParameters\fP
.RS 4
\fIanimatePathCoroutine\fP is a Coroutine used to animate a path as visual cue
.RE
.PP

.SS "float GazeGuidingPathPlayerSecondPath\&.animationDuration = 1\&.0f"

.PP

.PP
\fBParameters\fP
.RS 4
\fIanimationDuration\fP specifies the duration of the path animation
.RE
.PP

.SS "GameObject GazeGuidingPathPlayerSecondPath\&.arrow3DBinaerInstance\fR [private]\fP"

.PP

.PP
\fBParameters\fP
.RS 4
\fIarrow3DBinaerInstance\fP is an instance of an Arrow3D prefab used to render binary rotational 3DArrows<param> 
.RE
.PP

.SS "bool GazeGuidingPathPlayerSecondPath\&.arrow3DBinaerInstanceCreated = false\fR [private]\fP"

.PP

.PP
\fBParameters\fP
.RS 4
\fIarrow3DBinaerInstanceCreated\fP is a flag tracking whether an Arrow3DBinary instance has been created 
.RE
.PP

.SS "GameObject GazeGuidingPathPlayerSecondPath\&.arrow3DBinaerPrefab\fR [private]\fP"

.PP

.PP
\fBParameters\fP
.RS 4
\fIarrow3DBinaerPrefab\fP is a reference to an Arrow3D prefab used to render binary rotational 3DArrows
.RE
.PP

.SS "bool GazeGuidingPathPlayerSecondPath\&.Arrow3DBinearEnabled = true"

.PP

.PP
\fBParameters\fP
.RS 4
\fIArrow3DBinearEnabled\fP toggles 3D arrow indicating the direction a binary rotary switch should be turned in
.RE
.PP

.SS "bool GazeGuidingPathPlayerSecondPath\&.Arrow3DEnabled = true"

.PP

.PP
\fBParameters\fP
.RS 4
\fIArrow3DEnabled\fP toggles a 3D arrow indicating the direction an exact rotary switch should be turned in
.RE
.PP

.SS "GameObject GazeGuidingPathPlayerSecondPath\&.arrow3DInstance\fR [private]\fP"

.PP

.PP
\fBParameters\fP
.RS 4
\fIarrow3DInstance\fP is an instance of a Arrow3D prefab used to render continuous rotational 3DArrows<param> 
.RE
.PP

.SS "bool GazeGuidingPathPlayerSecondPath\&.arrow3DInstanceCreated = false\fR [private]\fP"

.PP

.PP
\fBParameters\fP
.RS 4
\fIarrow3DInstanceCreated\fP is a flag tracking whether an Arrow3D instance has been created 
.RE
.PP

.SS "GameObject GazeGuidingPathPlayerSecondPath\&.arrow3DPrefab\fR [private]\fP"

.PP

.PP
\fBParameters\fP
.RS 4
\fIarrow3DPrefab\fP is a reference to an Arrow3D prefab used to render continuous rotational 3DArrows
.RE
.PP

.SS "\fBGazeGuidingTarget\fP GazeGuidingPathPlayerSecondPath\&.currentTarget\fR [private]\fP"

.PP

.PP
\fBParameters\fP
.RS 4
\fIcurrentTarget\fP is the currently active \fBGazeGuidingTarget\fP for this \fBGazeGuidingPathPlayerSecondPath\fP
.RE
.PP

.SS "bool GazeGuidingPathPlayerSecondPath\&.DirectionArrowEnabled = true"

.PP

.PP
\fBParameters\fP
.RS 4
\fIDirectionArrowEnabled\fP toggles a directional arrow indicating a direction the player should turn to
.RE
.PP

.SS "bool GazeGuidingPathPlayerSecondPath\&.isAnimating = false\fR [private]\fP"

.PP

.PP
\fBParameters\fP
.RS 4
\fIisAnimating\fP tracks whether a path is currently being animated as visual cue
.RE
.PP

.SS "Material GazeGuidingPathPlayerSecondPath\&.lineMaterial"

.PP

.PP
\fBParameters\fP
.RS 4
\fIlineMaterial\fP is a Material used to render a path as visual cue
.RE
.PP

.SS "LineRenderer GazeGuidingPathPlayerSecondPath\&.lineRenderer\fR [private]\fP"

.PP

.PP
\fBParameters\fP
.RS 4
\fIlineRenderer\fP is a LineRenderer object used to render a path as visual cue
.RE
.PP

.SS "float GazeGuidingPathPlayerSecondPath\&.pathDisplayDistance = 5\&.0f"

.PP

.PP
\fBParameters\fP
.RS 4
\fIpathDisplayDistance\fP specifies the distance between the player and an animated path that is displayed as visual cue 
.RE
.PP

.SS "List<\fBGazeGuidingTarget\fP> GazeGuidingPathPlayerSecondPath\&.targets"

.PP

.PP
\fBParameters\fP
.RS 4
\fIDirectionArrowOnScreen\fP toggles the on screen display of the directional arrow
.RE
.PP

.PP
\fBParameters\fP
.RS 4
\fItargets\fP is a list of \fBGazeGuidingTarget\fP objects
.RE
.PP


.SH "Author"
.PP 
Generated automatically by Doxygen for NPP VR Simulation from the source code\&.


Assets/Skripte/GazeGuidingPath/GazeGuidingPathPlayerSecondPath.cs
using UnityEngine;
using System.Collections;
using System.Collections.Generic;
using UnityEngine.UI;

/// <summary>
/// This class implements logic for gaze guiding in a virtual environment.
/// <summary>
public class GazeGuidingPathPlayerSecondPath : MonoBehaviour
{
/// <param name="DirectionArrowEnabled"> toggles a directional arrow indicating a direction the player should turn to</param>
    public bool DirectionArrowEnabled = true;    
    /// <param name="Arrow3DEnabled"> toggles a 3D arrow indicating the direction an exact rotary switch should be turned in</param>
    public bool Arrow3DEnabled = true;
    /// <param name="Arrow3DBinearEnabled"> toggles 3D arrow indicating the direction a binary rotary switch should be turned in</param>
    public bool Arrow3DBinearEnabled = true;
    /// <param name="DirectionArrowOnScreen"> toggles the on screen display of the directional arrow</param>
    
    /// <param name="targets"> is a list of GazeGuidingTarget objects</param>
    public List<GazeGuidingTarget> targets;
    /// <param name="currentTarget"> is the currently active GazeGuidingTarget for this GazeGuidingPathPlayerSecondPath</param>
    private GazeGuidingTarget currentTarget;
    /// <param name="pathDisplayDistance">  specifies the distance between the player and an animated path that is displayed as visual cue </param>
    public float pathDisplayDistance = 5.0f;
    /// <param name="animationDuration"> specifies the duration of the path animation</param>
    public float animationDuration = 1.0f;
    /// <param name="lineMaterial">  is a Material used to render a path as visual cue</param>
    public Material lineMaterial; 
    /// <param name="lineRenderer"> is a LineRenderer object used to render a path as visual cue</param>
    private LineRenderer lineRenderer;
    /// <param name="animatePathCoroutine"> is a Coroutine used to animate a path as visual cue</param>
    private Coroutine animatePathCoroutine;
    /// <param name="isAnimating"> tracks whether a path is currently being animated as visual cue</param>
    private bool isAnimating = false;
    
    /// <param name="arrow3DPrefab"> is a reference to an Arrow3D prefab used to render continuous rotational 3DArrows</param>    
    private GameObject arrow3DPrefab; 
    /// <param name="arrow3DInstance"> is an instance of a Arrow3D prefab used to render continuous rotational 3DArrows<param>

    private GameObject arrow3DInstance; 
    /// <param name="arrow3DBinaerPrefab"> is a reference to an Arrow3D prefab used to render binary rotational 3DArrows</param>
    
    private GameObject arrow3DBinaerPrefab; 
    /// <param name="arrow3DBinaerInstance"> is an instance of an Arrow3D prefab used to render binary rotational 3DArrows<param>
    private GameObject arrow3DBinaerInstance; // Instance of the Arrow3D prefab
    

    /// <summary>
    /// This method initialises the GazeGuidingPathPlayerSecondPath. 
    /// </summary>
    void Start()
    {
        
        lineRenderer = gameObject.AddComponent<LineRenderer>();
        lineRenderer.startWidth = 0.03f;
        lineRenderer.endWidth = 0.03f;

        lineRenderer.positionCount = 0;
        lineRenderer.material = lineMaterial; 

        // Set texture mode to Tile
        lineRenderer.textureMode = LineTextureMode.Tile;

        // Set texture scale
        lineRenderer.textureScale = new Vector2(2.36f, 0.57f);
        
        // Load the Arrow3D prefab
        arrow3DPrefab = Resources.Load<GameObject>("Prefabs/Arrow3D");
        arrow3DBinaerPrefab = Resources.Load<GameObject>("Prefabs/Arrow3DBinaer");
        
        targets = new List<GazeGuidingTarget>(FindObjectsOfType<GazeGuidingTarget>());
    }
    
    /// <param name="arrow3DInstanceCreated"> is a flag tracking whether an Arrow3D instance has been created </param>
    private bool arrow3DInstanceCreated = false; 
    /// <param name="arrow3DBinaerInstanceCreated"> is a flag tracking whether an Arrow3DBinary instance has been created </param>    
    private bool arrow3DBinaerInstanceCreated = false; 
    /// <summaQry>
    /// This method updates all active gaze-guiding tools.
    /// </summary>
    void Update()
    {   
        if (currentTarget != null)
        {
            if (Time.frameCount % 2 == 0 && DirectionArrowEnabled)
            {
                float distance = Vector3.Distance(transform.position, currentTarget.transform.position);
                if (distance < pathDisplayDistance)
                {
                    lineRenderer.positionCount = 0;
                    isAnimating = false;
                    if (animatePathCoroutine != null)
                    {
                        StopCoroutine(animatePathCoroutine);
                        animatePathCoroutine = null;
                    }
                    
                    // Instantiate Arrow3D prefab if not already instantiated
                    if (!arrow3DInstanceCreated)
                    {
                        Arrow3D();
                        arrow3DInstanceCreated = true;
                    }

                    if (!arrow3DBinaerInstanceCreated)
                    {
                        Arrow3DBinaer();
                        arrow3DBinaerInstanceCreated = true;
                    }
                }
                else
                {
                    
                    // Remove the Arrow3D instance if the distance is too high
                    RemoveArrow3D();
                    if (!isAnimating)
                    {
                        animatePathCoroutine = StartCoroutine(AnimatePath(currentTarget.transform.position));
                    }
                }
            }
            else if(!DirectionArrowEnabled)
            {
                if (animatePathCoroutine != null)
                {
                    StopCoroutine(animatePathCoroutine);
                    animatePathCoroutine = null;
                }
                isAnimating = false;
                if (animatePathCoroutine != null)
                {
                    StopCoroutine(animatePathCoroutine);
                    animatePathCoroutine = null;
                }
                lineRenderer.positionCount = 0;
            }
        }
    }

    /// <summary>
    /// This method adds a continuous rotating 3D arrow that is rendered above an exact rotary switch to indicate the direction in which the player is supposed to turn the switch.
    /// </summary>
    public void Arrow3D()
    {
        if (Arrow3DEnabled && currentTarget.isTypeOf == GazeGuidingTarget.TargetType.Genau)
        {
            if (arrow3DInstance == null)
            {
                arrow3DInstance = Instantiate(arrow3DPrefab, 
                    currentTarget.transform.position + new Vector3(0, 0.1f, 0), Quaternion.identity);
            }
            
        }
    }

    /// <summary>
    /// This method adds a binary rotating 3D arrow rendered above a binary rotary switch to indicate the direction in which the player is supposed to turn the switch.
    /// </summary>    
    public void Arrow3DBinaer()
    {
        if (Arrow3DBinearEnabled && currentTarget.isTypeOf == GazeGuidingTarget.TargetType.Binaer)
        {
            if (arrow3DBinaerInstance == null)
            {
                arrow3DBinaerInstance = Instantiate(arrow3DBinaerPrefab, 
                    currentTarget.transform.position + new Vector3(0, 0.1f, 0), Quaternion.identity);
            }
            
        }
    }

    /// <summary>
    /// This method removes a rotating 3D arrow or 3DBinary arrow that is currently being rendered above an exact or a binary rotary switch.
    /// </summary>    
    public void RemoveArrow3D()
    {
        if (arrow3DInstance != null)
        {
            Destroy(arrow3DInstance);
            arrow3DInstance = null;
        }

        arrow3DInstanceCreated = false;

        if (arrow3DBinaerInstance != null)
        {
            Destroy(arrow3DBinaerInstance);
            arrow3DBinaerInstance = null;
        }
        
        arrow3DBinaerInstanceCreated = false;
    }
    
    // deprecated, not included in documentation

    public void TriggerTarget(GazeGuidingTarget target, bool Flip3DArrow = false)
    {
        ClearLine();
        currentTarget = target;
        if (Flip3DArrow)
        {
            arrow3DPrefab.GetComponent<Rotate3DArrow>().flipDirection = true;
        }
        else
        {
            arrow3DPrefab.GetComponent<Rotate3DArrow>().flipDirection = false;
        }
    }

    
    
    //Aus einem anderen Skript aufrufbar mit z.B.:
    //GazeGuidingPathPlayerSecondPath gazeGuidingPathPlayerSecondPath = FindObjectOfType<GazeGuidingPathPlayerSecondPath>();
    //gazeGuidingPathPlayerSecondPath.TriggerTargetNAME("WP1RPM", GazeGuidingTarget.TargetType.Anzeige);
    //dadruch wird dann das gazeGuiding auf WP1RPM-anzeige gesetzt
    //die restlichen eingaben sind die GameObjektnamen mit dem GazeGuidingTarget skript
    

    /// <summary>
    /// This method sets the current target of a GazeGuidingPathPlayerSecondPath instance.
    /// </summary>
    /// <param name="targetName">  contains the name of a rotary switch that is the new target for the GazeGuidingPathPlayerSecondPath </param>
    /// <param name="type"> is an Enum specifying the type of rotary switch passed in targetName </param>
    /// <param name="Flip3DArrow"> toggles whether 3D arrows rotate clockwise </param>
    public void TriggerTargetNAME(string targetName,  GazeGuidingTarget.TargetType type, bool Flip3DArrow = false)
    {
        ClearLine();
        currentTarget = targets.Find(t => t.name == targetName && t.isTypeOf == type);
        if (currentTarget != null)
        {
            if (type == GazeGuidingTarget.TargetType.Genau)
            {
                if (Flip3DArrow)
                {
                    arrow3DPrefab.GetComponent<Rotate3DArrow>().flipDirection = true;
                }
                else
                {
                    arrow3DPrefab.GetComponent<Rotate3DArrow>().flipDirection = false;
                }
            }else if (type == GazeGuidingTarget.TargetType.Binaer)
            {
                if (Flip3DArrow)
                {
                    arrow3DBinaerPrefab.GetComponent<Rotate3DArrowBinaer>().flipDirection = true;
                }
                else
                {
                    arrow3DBinaerPrefab.GetComponent<Rotate3DArrowBinaer>().flipDirection = false;
                }
            }
        }
        else
        {
            Debug.LogWarning($"Target with name {targetName} and type {type} not found.");
        }
    }
    
    /// <summary>
    /// This method clears the scene of all visual aids that are currently being rendered.
    /// </summary>    
    public void ClearLine()
    {
        // Remove the Arrow3D instance
        RemoveArrow3D();
        
        currentTarget = null;
        isAnimating = false;
        if (animatePathCoroutine != null)
        {
            StopCoroutine(animatePathCoroutine);
            animatePathCoroutine = null;
        }
        lineRenderer.positionCount = 0;
    }

    /// <summary>
    /// This method animates a path of arrows guiding the player towards the component set as currentTarget of the GazeGuidingPathPlayerSecondPath.
    /// </summary>
    /// <param name="targetPosition"> is a Vector3 position of the endpoint of the path </param>
    private IEnumerator AnimatePath(Vector3 targetPosition)
    {
        isAnimating = true;
        while (true)
        {
            float elapsedTime = 0f;

            while (elapsedTime < animationDuration)
            {
                elapsedTime += Time.deltaTime;
                float t = Mathf.Clamp01(elapsedTime / animationDuration);
                Vector3 currentPoint = Vector3.Lerp(transform.position, targetPosition, t);

                lineRenderer.positionCount = 2;
                lineRenderer.SetPosition(0, transform.position);
                lineRenderer.SetPosition(1, currentPoint);

                yield return null;
            }

            lineRenderer.positionCount = 2;
            lineRenderer.SetPosition(0, transform.position);
            lineRenderer.SetPosition(1, targetPosition);

            float waitTime = 0.5f;
            while (waitTime > 0)
            {
                lineRenderer.SetPosition(0, transform.position);
                waitTime -= Time.deltaTime;
                yield return null;
            }
        }
    }
}
